examples/todomvc/test/todomvc/core_test.cljc

```diff
@@ -0,0 +1,126 @@
+      (is (empty? @visible-todos))
+      (is (= false (boolean @all-complete?)))
+      (is (= 0 @completed-count))
+      (is (= [0 0] @footer-counts))
+
+      (rf/dispatch [:add-todo "write first test"])
+      (is (= 1 (count @todos) (count @visible-todos) (count @sorted-todos)))
+      (is (= 0 @completed-count))
+      (is (= [1 0] @footer-counts))
+      (is (= {:id 1, :title "write first test", :done false}
+             (first @todos)))
+
+      (rf/dispatch [:add-todo "write second teXt"])
+      (is (= 2 (count @todos) (count @visible-todos) (count @sorted-todos)))
+      (is (= 0 @completed-count))
+      (is (= [2 0] @footer-counts))
+      (is (= {:id 2, :title "write second teXt", :done false}
+             (second @todos)))
+
+      (rf/dispatch [:save 2 "write second test"])
+      (is (= 2 (count @todos) (count @visible-todos) (count @sorted-todos)))
+      (is (= 0 @completed-count))
+      (is (= [2 0] @footer-counts))
+      (is (= {:id 2, :title "write second test", :done false}
+             (second @todos)))
+
+      (rf/dispatch [:toggle-done 1])
+      (is (= 2 (count @todos) (count @visible-todos) (count @sorted-todos)))
+      (is (= 1 @completed-count))
+      (is (= [1 1] @footer-counts))
+      (is (= {:id 1, :title "write first test", :done true}
+             (first @todos)))
+
+      (rf/dispatch [:set-showing :active])
+      (is (= :active @showing))
+      (is (= 2 (count @todos) (count @sorted-todos)))
+      (is (= 1 (count @visible-todos)))
+      (is (= 1 @completed-count))
+      (is (= [1 1] @footer-counts))
+      (is (= {:id 2, :title "write second test", :done false}
+             (first @visible-todos)))
+
+      (rf/dispatch [:set-showing :done])
+      (is (= :done @showing))
+      (is (= 2 (count @todos) (count @sorted-todos)))
+      (is (= 1 (count @visible-todos)))
+      (is (= 1 @completed-count))
+      (is (= [1 1] @footer-counts))
+      (is (= {:id 1, :title "write first test", :done true}
+             (first @visible-todos)))
+
+      (rf/dispatch [:toggle-done 2])
+      (is (= true (boolean @all-complete?))))))
+
+
+(deftest basic--async
+  (rf-test/run-test-async
+    (rf/dispatch-sync [:initialise-db])
+
+    (let [showing         (rf/subscribe [:showing])
+          sorted-todos    (rf/subscribe [:sorted-todos])
+          todos           (rf/subscribe [:todos])
+          visible-todos   (rf/subscribe [:visible-todos])
+          all-complete?   (rf/subscribe [:all-complete?])
+          completed-count (rf/subscribe [:completed-count])
+          footer-counts   (rf/subscribe [:footer-counts])]
+      (is (= :all @showing))
+      (is (empty? @sorted-todos))
+      (is (empty? @todos))
+      (is (empty? @visible-todos))
+      (is (= 0 @completed-count))
+
+      (rf/dispatch [:add-todo "write first test"])
+      (rf-test/wait-for [:add-todo]
+        (is (= [{:id 1, :title "write first test", :done false}] @todos))
+
+        (rf/dispatch [:add-todo "write second teXt"])
```

Comment:
One thing that could be good to add would be a testing dispatching one event which async dispatches a second, and waiting for the second. But don't worry right now, as not sure where/how we'll merge these tests.